Packet++/src/TcpReassembly.cpp

```diff
@@ -473,34 +473,40 @@ void TcpReassembly::handleFinOrRst(TcpReassemblyData* tcpReassemblyData, int8_t
 
 void TcpReassembly::checkOutOfOrderFragments(TcpReassemblyData* tcpReassemblyData, int8_t sideIndex, bool cleanWholeFragList)
 {
+	if (m_ProcessingOutOfOrder)
+	{
+		return;
+	}
+
+	m_ProcessingOutOfOrder = true;
 	bool foundSomething = false;
 
 	do
 	{
 		PCPP_LOG_DEBUG("Starting first iteration of checkOutOfOrderFragments - looking for fragments that match the current sequence or have smaller sequence");
 
-		int index = 0;
+		PointerVector<TcpFragment>::VectorIterator it;
```

Comment:
I don't really like how the iterator is declared here and then used all throughout the function. Sure perhaps it is more efficient (not really sure how much tho, since its stack allocated), but IMO it makes it harder to read the function code as I have to scroll up to figure out what that iterator is and why it is not declared at the loop usage, but somewhere else.

So far, it is only used twice. both times to do full loops through `tcpFragmentList`.
- Once on L498 through L568
- Again on L591 through L607, where the previous value is overwritten by a `begin()` call.

IMO, the iterator declaration should be pulled next to the L498 loop and the L591 loop be converted to a `for-loop`

Reply:
Updated in this commit: https://github.com/seladb/PcapPlusPlus/pull/1346/commits/64562380a1b754123e8f3e461c5a6c1771bb8a11